py_experimenter/experimenter.py

```diff
@@ -29,58 +29,101 @@ def __init__(self,
+        :type name: str, optional
+        :raises InvalidConfigError: If either the experiment or database configuration are missing mandatory information.
+        :raises ValueError: If an unsupported or unknown database connection provider is given.
         """
-        self._config = utils.load_config(config_file)
-        self._credential_path = credential_path
-        if not PyExperimenter._valid_configuration(self._config, credential_path):
+        self.config = utils.load_config(config_file)
+        self.credential_path = credential_path
+        if not PyExperimenter._is_valid_configuration(self.config, credential_path):
             raise InvalidConfigError('Invalid configuration')
 
         if table_name is not None:
-            self._config.set('PY_EXPERIMENTER', 'table', table_name)
+            self.config.set('PY_EXPERIMENTER', 'table', table_name)
         if database_name is not None:
-            self._config.set('PY_EXPERIMENTER', 'database', database_name)
+            self.config.set('PY_EXPERIMENTER', 'database', database_name)
         self.name= name
 
-        self._config_file = config_file
-        self.timestamp_on_result_fields = utils.timestamps_for_result_fields(self._config)
+        self.config_file = config_file
+        self.timestamp_on_result_fields = utils.timestamps_for_result_fields(self.config)
 
-        if self._config['PY_EXPERIMENTER']['provider'] == 'sqlite':
-            self._dbconnector = DatabaseConnectorLITE(self._config)
-        elif self._config['PY_EXPERIMENTER']['provider'] == 'mysql':
-            self._dbconnector = DatabaseConnectorMYSQL(self._config, credential_path)
+        if self.config['PY_EXPERIMENTER']['provider'] == 'sqlite':
+            self.dbconnector = DatabaseConnectorLITE(self.config)
+        elif self.config['PY_EXPERIMENTER']['provider'] == 'mysql':
+            self.dbconnector = DatabaseConnectorMYSQL(self.config, credential_path)
         else:
             raise ValueError('The provider indicated in the config file is not supported')
 
         logging.info('Initialized and connected to database')
 
     def set_config_value(self, section_name: str, key: str, value: str) -> None:
         """
-        Sets a value in the configuration file. If the section does not exist, it is created.
-        :param section_name: Name of the section in the configuration file.
-        :param key: Name of the key in the configuration file.
-        :param value: Value to set.
+        Modifies the experiment configuration so that within the given `section_name` the value of the given `key` is overridden, 
+        or created if it is not existing beforehand.
+
+        :param section_name: The name of the section of the experiment configuration in which a value should be set.
+        :type section_name: str
+        :param key: The name of the key within the given section of the experiment configuration in which a value should be set.
```

Comment:
The name of the key identifying the property within the given section whose value should be set.